Dev/jdsequeira/us vbs22 20 google flights api

DieS06 opened the pull request.

DieS06 committed:
Flight Entity

Create of entity, repository, service and controller for flight object.

DieS06 committed:
Removing Request to API

DieS06 committed:
RestTemplate for GoogleFlights

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Google Flights

Search Query, Entities, etc.

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Temporal Commit

DieS06 committed:
Fixing

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Back-end Flight Done

DieS06 committed:
Update GoogleFlightsRequestController.java

Eliminated Api Key param from controller.

DieS06 committed:
Fixing Country Code

DieS06 committed:
Add Airport Entity

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-Google-Flights-API

DieS06 committed:
Fixing Endpoint

DieS06 committed:
URI Construction Fixed

DieS06 committed:
Merge branch 'development' into dev/jdsequeira/US-VBS22-20-GoogleFlightsAPI

nefi-cenfo commented on demo/src/main/java/com/project/demo/entity/Airport.java:
```diff
@@ -0,0 +1,49 @@
+    @Id
+    private Integer airport_id;
+    private String name;
+    private String id;
```
What is the propose of this property?

DieS06 replied:
the airport_id is the number identifier and the id is the code or airport that google flights response throws.

nefi-cenfo commented on demo/src/main/java/com/project/demo/logic/request/GoogleFlightsRequestService.java:
```diff
@@ -0,0 +1,71 @@
+    private void addQueryParameters(UriComponentsBuilder builder, GoogleFlightsRequestEntity query) throws Exception {
+            if (query.getType() == 1 && query.getReturn_date() != null) {
+                builder.queryParam("return_date", query.getReturn_date());
+            } else if (query.getType() == 2 && query.getReturn_date() == null) {
```
Beware of the else if block is doing nothing.

nefi-cenfo commented on demo/src/main/java/com/project/demo/logic/request/OpenAIService.java:
```diff
@@ -21,80 +21,81 @@
-        this.restTemplate = restTemplate;
-        this.objectMapper = objectMapper;
-    }
+//    @Value("${openai.api.key}")
```
Please uncomment this lines of code before merging.

nefi-cenfo commented on demo/src/main/java/com/project/demo/rest/request/GoogleFlightsRequestController.java:
```diff
@@ -0,0 +1,40 @@
+import java.util.Locale;
+
+@RestController
+@RequestMapping("/api/flights")
```
What is the difference between this and the above one?

DieS06 committed:
Comments Resolved

DieS06 committed:
Fixing Service and Controller

Methods were wrongly set up as Object return and they need to be strings to capture the API response.

mur-tz reviewed (approved):
For the next iteration, we may merge the  two controllers on this PR, since they are similar and I think you only use one controller for the request one
